File: jna/src/main/java/dark/update_args.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

public class update_args extends Structure {
    int batch;
    float learning_rate;
    float momentum;
    float decay;
    int adam;
    float B1;
    float B2;
    float eps;
    int t;
    public update_args() {
        super();
    }

    public update_args(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("batch","learning_rate","momentum","decay","adam","B1","B2","eps","t");
    }
}

File: jna/src/main/java/dark_kotlin/update_args.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure

class update_args : Structure {

    @JvmField val batch = 0
    @JvmField val learning_rate = 0f
    @JvmField val momentum = 0f
    @JvmField val decay = 0f
    @JvmField val adam = 0
    @JvmField val B1 = 0f
    @JvmField val B2 = 0f
    @JvmField val eps = 0f
    @JvmField val t = 0

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("batch", "learning_rate", "momentum", "decay", "adam", "B1", "B2", "eps", "t")
    }
}
